wordpress-mobile/WordPress-Android
# [Update Detekt Version] Update Detekt to 1.16.0

Parent: #16892
Closes: #16894

This PR:
- Updates Detekt to `1.16.0` (from `1.15.0`) (3efb9e86bf0c69bdf03560a06e0df37432266a5e).
- Resolves/Suppresses new warnings that started appearing due to the fact that the below rules are now enabled by default (previously disabled by default):
    - `PrintStackTrace` (exceptions)
    - `SwallowedException` (exceptions)
    - `ImplicitDefaultLocale` (potential-bugs)
    - `SerialVersionUIDInSerializableClass` (style)
    - `UnusedPrivateMember` (style)
    - `VarCouldBeVal` (style)

PS: I recommend reviewing this PR commit-by-commit.

-----

To test:

- Verify that all the CI checks are successful (especially the `detekt` check).
- Smoke test the app and verify everything works as expected.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
